packages/backend/services/crm/deal.ts

```diff
@@ -48,16 +51,39 @@ const dealService = new DealService(
                             'hs_is_closed_won',
                             'hs_createdate',
                         ];
+                        const validAssociations = [...associations].filter((item) =>
+                            isValidAssociationTypeRequestedByUser(item)
+                        );
+                        const invalidAssociations = [...associations].filter(
+                            (item) =>
+                                item !== 'undefined' && item !== 'null' && !isValidAssociationTypeRequestedByUser(item)
+                        );
+                        let url;
+                        if (validAssociations.length > 0) {
+                            url = `https://api.hubapi.com/crm/v3/objects/deals/${dealId}?associations=${validAssociations}&properties=${formattedFields}`;
+                        } else {
+                            url = `https://api.hubapi.com/crm/v3/objects/deals/${dealId}?properties=${formattedFields}`;
+                        }
+
```

Comment:
@manish-singh-bisht there is better way to write this, you might change to follow these or the enum style as suggesting below, in every files under `services/crm` : 

```suggestion
      const url = `https://api.hubapi.com/crm/v3/objects/deals/${dealId}?properties=${formattedFields}` +
      (validAssociations.length > 0 && `&associations=${validAssociations}`);

```
also usually many orgs follow to have a enums for baseUrl. so in future you also might encounter something like this aswell, one of the reason for using this style, what if the third party for eg hubspot change there api versioning or baseUrl. we will need to change it in everyPlace following style ensures that we have single source of truth. 

```
enum baseUrl {
     HUBSPOT: 'https://api.hubapi.com/crm/v3'
}
```

const url = `${baseUrl.HUBSPOT.toString()}${path}${params}`

you can change it to use combination of both.

Reply:
@Nabhag8848  did the url in  a better way as you suggested. Thank you for that.

for the base url thing,
for example: for hubspot the baseUrl will be "https://api.hubapi.com" because one url say for "company" object is "https://api.hubapi.com/crm/v3/objects/companies/${companyId}" and one for "user" object  is "https://api.hubapi.com/settings/v3/users"

so the versioning happens after we specify whether it is from "object" or from "setting". the base url will be able to justify itself partially.